modules/broker-amqp/src/main/java/org/wso2/broker/amqp/codec/InMemoryMessageAggregator.java

```diff
@@ -66,6 +79,19 @@ public void headerFrameReceived(FieldTable headers, FieldTable properties, long
         metadata.setProperties(properties);
         metadata.setHeaders(headers);
         message = new Message(metadata);
+        trace(metadata);
+    }
+
+    private void trace(Metadata metadata) {
+        if (MessageTracer.isTraceEnabled()) {
+            List<TraceField> traceFields = new ArrayList<>();
+            FieldValue fieldValue = metadata.getProperty(Metadata.CORRELATION_ID);
+            if (Objects.nonNull(fieldValue)) {
+                TraceField field = new TraceField(CORRELATION_ID_FIELD_NAME, fieldValue.getValue());
+                traceFields.add(field);
+            }
+            MessageTracer.trace(metadata, INCOMING_MESSAGE_MAPPED, traceFields);
```

Comment:
This will have some impact on the netty even loop since logging involves some I/O. Maybe we can use async logginc to overcome this. WDYT?

https://logging.apache.org/log4j/2.x/manual/async.html

Reply:
+1 I've created an issue to track this. #174